src/adapters/bidderFactory.js

```diff
@@ -493,19 +497,55 @@ export const processBidderRequests = hook('async', function (spec, bids, bidderR
         );
         break;
       case 'POST':
-        ajax(
-          request.url,
-          {
-            success: onSuccess,
-            error: onFailure
-          },
-          typeof request.data === 'string' ? request.data : JSON.stringify(request.data),
-          getOptions({
-            method: 'POST',
-            contentType: 'text/plain',
-            withCredentials: true
-          })
-        );
+        const enableGZipCompression = bidderSettings.get(spec.code, 'endpointCompression');
+        const debugMode = getParameterByName(DEBUG_MODE).toUpperCase() === 'TRUE' || debugTurnedOn();
+
+        if (enableGZipCompression && debugMode) {
+          logWarn(`Skipping GZIP compression for ${spec.code} as debug mode is enabled`);
+        }
+
+        if (enableGZipCompression && !debugMode && isGzipCompressionSupported()) {
+          compressDataWithGZip(request.data).then(compressedPayload => {
+            const url = new URL(request.url, window.location.origin);
+            if (!url.searchParams.has('gzip')) {
+              url.searchParams.set('gzip', '1');
+            }
+
+            ajax(
+              url.href,
+              {
+                success: onSuccess,
+                error: onFailure
+              },
+              compressedPayload,
+              getOptions({
+                method: 'POST',
+                contentType: 'text/plain',
+                // adding below header creates a CORS preflight request so we should not pass it
+                // instead of adding this header we can pass a query parmeter (gzip=1) to let server know that the payload is compressed
+                // TODO: update URL to include query parameter gzip=1
+                // customHeaders: {
+                //  'Content-Encoding': 'gzip',
+                // },
+                withCredentials: true
+              })
+            );
+          });
+        } else {
+          ajax(
+            request.url,
+            {
+              success: onSuccess,
+              error: onFailure
+            },
+            typeof request.data === 'string' ? request.data : JSON.stringify(request.data),
+            getOptions({
+              method: 'POST',
+              contentType: 'text/plain',
+              withCredentials: true
+            })
+          );
+        }
```

Comment:
This repeats almost the same logic twice. It could be refactored to something like `getData().then(callAjax)`.